### PR TITLE
Introduce Spring Session MongoDB.

### PR DESCRIPTION
**Pull request by gregturn**
* Migrate the module's code back into this project.
* Fold the documentation in.
* Update to current Gradle conventions.
* Reformat to match styling.

### CODE REVIEW

**Comment by gregturn** [comment]
Things @vpavic for checking out my PR. I'll make changes.

File: gradle/dependency-management.gradle [hunk 1]
```diff
@@ -28,9 +33,21 @@ dependencyManagement {
 		dependency 'mysql:mysql-connector-java:8.0.26'
 		dependency 'org.apache.derby:derby:10.14.2.0'
 		dependency 'org.assertj:assertj-core:3.20.2'
+        dependency 'org.hamcrest:hamcrest:2.1'
 		dependency 'org.hsqldb:hsqldb:2.5.1'
 		dependency 'org.mariadb.jdbc:mariadb-java-client:2.7.4'
 		dependency 'org.mockito:mockito-core:3.11.2'
+        dependencySet(group: 'org.mockito', version: '3.10.0') {
```
[the remark this answers is not in the document]
Comment (gregturn) [reply gradle/dependency-management.gradle:40]: Fixed.

File: gradle/dependency-management.gradle [hunk 2]
```diff
@@ -15,11 +16,15 @@ dependencyManagement {
 		}
 
 		dependency 'org.aspectj:aspectjweaver:1.9.7'
+		dependency 'org.aspectj:aspectjweaver:1.9.6'
```
[the remark this answers is not in the document]
Comment (gregturn) [reply gradle/dependency-management.gradle:19]: Fixed.

File: spring-session-mongodb/pom.xml [hunk 3]
```diff
@@ -0,0 +1,986 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (gregturn) [reply spring-session-mongodb/pom.xml:1]: Fixed.

File: gradle/dependency-management.gradle [hunk 4]
```diff
@@ -15,11 +16,15 @@ dependencyManagement {
...
+		dependency 'ch.qos.logback:logback-core:1.2.3'
+		dependency 'com.google.code.findbugs:jsr305:3.0.2'
 		dependency 'com.h2database:h2:1.4.200'
 		dependency 'com.ibm.db2:jcc:11.5.6.0'
 		dependency 'com.microsoft.sqlserver:mssql-jdbc:9.4.0.jre8'
 		dependency 'com.oracle.database.jdbc:ojdbc8:21.1.0.0'
 		dependency 'com.zaxxer:HikariCP:3.4.5'
+		dependency 'de.flapdoodle.embed:de.flapdoodle.embed.mongo:2.2.0'
```
[the remark this answers is not in the document]
Comment (gregturn) [reply gradle/dependency-management.gradle:27]: Done.

File: settings.gradle [hunk 5]
```diff
@@ -17,6 +17,7 @@ include 'spring-session-data-redis'
 include 'spring-session-docs'
 include 'spring-session-hazelcast'
 include 'spring-session-jdbc'
+include 'spring-session-mongodb'
```
[the remark this answers is not in the document]
Comment (gregturn) [reply settings.gradle:20]: Done.

File: spring-session-docs/src/docs/asciidoc/index.adoc [hunk 6]
```diff
@@ -140,13 +148,11 @@ While convenient, this approach was not sustainable long-term as more features a
 
 Starting with Spring Session 2.0, the project has been split into Spring Session Core module and several other modules that carry `SessionRepository` implementations and functionality related to the specific data store.
 Users of Spring Data should find this arrangement familiar, with Spring Session Core module taking a role equivalent to Spring Data Commons and providing core functionalities and APIs, with other modules containing data store specific implementations.
-As part of this split, the Spring Session Data MongoDB and Spring Session Data GemFire modules were moved to separate repositories.
+As part of this split, the Spring Session Data GemFire module was moved to a separate repository.
```
[the remark this answers is not in the document]
Comment (gregturn) [reply spring-session-docs/src/docs/asciidoc/index.adoc:151]: Done.

File: spring-session-mongodb/src/main/asciidoc/index.adoc [hunk 7]
```diff
@@ -0,0 +1,139 @@
+
```
[the remark this answers is not in the document]
Comment (gregturn) [reply spring-session-mongodb/src/main/asciidoc/index.adoc:1]: Done

File: spring-session-mongodb/spring-session-mongodb.gradle [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+	compile ("org.springframework.data:spring-data-mongodb") {
+		exclude group: "org.mongodb", module: "mongo-java-driver"
+		exclude group: "org.slf4j", module: "jcl-over-slf4j"
+	}
+
+	// MongoDB dependencies
+
+	optional "org.mongodb:mongodb-driver-core"
+	testCompile "org.mongodb:mongodb-driver-sync"
+	testCompile ("de.flapdoodle.embed:de.flapdoodle.embed.mongo") {
```
[the remark this answers is not in the document]
Comment (gregturn) [reply spring-session-mongodb/spring-session-mongodb.gradle]: Done

**Comment by gregturn** [comment]
I've applied all the changes requested @vpavic. I squashed them together. Feel free to offer more comments.

**Comment by vpavic** [comment]
Hey @gregturn, I'll try to take another look at this by the end of the week.

File: spring-session-samples/spring-session-sample-boot-mongodb-reactive/spring-session-sample-boot-mongodb-reactive.gradle [hunk 9]
```diff
@@ -0,0 +1,13 @@
+apply plugin: 'io.spring.convention.spring-sample-boot'
+
+dependencies {
+	compile project(':spring-session-data-mongodb')
+	compile "org.springframework.boot:spring-boot-starter-webflux"
+	compile "org.springframework.boot:spring-boot-starter-thymeleaf"
+	compile "org.springframework.boot:spring-boot-starter-data-mongodb-reactive"
+    compile "de.flapdoodle.embed:de.flapdoodle.embed.mongo"
```
Comment (eleftherias) [inline spring-session-samples/spring-session-sample-boot-mongodb-reactive/spring-session-sample-boot-mongodb-reactive.gradle:8]: I added an embedded MongoDB to both samples. 
The alternative is to expect the user to have a local MongoDB instance on their machine before running the sample.  In that case we would need to update the integration tests for the samples to use test containers.

**Review by eleftherias — COMMENTED** [review]
@gregturn I added commits to rebase on main and update the documentation to use Antora.
I also fixed the samples, I added an inline comment about the change that I made, let me know if you prefer an alternative approach.

**Pull request closed without merging**